merkle/maphasher/maphasher.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package maphasher provides hashing for maps.
+package maphasher
+
+import (
+	"crypto"
+	"fmt"
+)
+
+// Domain separation prefixes
+const (
+	leafHashPrefix = 0
+	nodeHashPrefix = 1
+)
+
+// Default is a SHA256 based MapHasher for maps.
+var Default = New(crypto.SHA256)
+
+// MapHasher implements a sparse merkel tree hashing algorithm. For testing only.
```

Comment:
I thought we should make it clearer that this is an insecure test hasher. If it's documented as the "default" people might use it without realizing.